code/modules/research/xenoarchaeology/machinery/analysis_base.dm

```diff
@@ -51,67 +49,66 @@
 	..()
 	if (stat & NOPOWER && scan_process)
 		stop()
-
 	else
 		update_icon()
 
 /obj/machinery/anomaly/process()
-	//not sure if everything needs to heat up, or just the GLPC
-	var/datum/gas_mixture/env = loc.return_air()
-	var/environmental_temp = env.temperature
+	//First we deal with the machine's task
 	if(scan_process)
-		// Shouldn't be reachable, still can't hurt.
-		if(stat & NOPOWER)
+		if (stat & (NOPOWER|BROKEN))
 			stop()
-
-		if(scan_process++ > target_scan_ticks)
-			FinishScan()
-		else if(temperature > XENOARCH_MAX_TEMP)
-			visible_message("<span class='notice'>[bicon(src)] shuts down from the heat!</span>")
-			scan_process = 0
-		else if(temperature > XENOARCH_SAFETY_TEMP && prob(10))
-			visible_message("<span class='notice'>[bicon(src)] bleets plaintively.</span>")
-
-		//show we're busy
-		if(prob(5))
-			visible_message("<span class='notice'>[bicon(src)] [pick("whirrs","chuffs","clicks")][pick(" quietly"," softly"," sadly"," excitedly"," energetically"," angrily"," plaintively")].</span>")
-
-		use_power = 2
-
+		else
+			use_power = 2
```

Comment:
Magic numbers bad. It was not until I read line 80 that I figured that (1 == idle) and (2 == active).

Use the defines from machinery.dm

```
#define MACHINE_POWER_USE_NONE   0
#define MACHINE_POWER_USE_IDLE   1
#define MACHINE_POWER_USE_ACTIVE 2
```